src/core/algorithms/statistics/data_stats.cpp

```diff
@@ -606,6 +623,295 @@ Statistic DataStats::GetAvgNumberOfChars(size_t index) const {
+    }
+    std::ostringstream oss;
+    for (auto const &word : words) {
+        oss << word << ' ';
+    }
+
+    std::string const &temp = oss.str();
+    std::byte const *res = string_type.MakeValue(temp);
+
+    return Statistic(res, &string_type, false);
+}
+
+Statistic DataStats::GetMinWords(size_t index) const {
+    if (all_stats_[index].min_words.HasValue()) return all_stats_[index].min_words;
+
+    mo::TypedColumnData const &col = col_data_[index];
+    if (col.GetTypeId() != +mo::TypeId::kString) return {};
+
+    mo::IntType int_type;
+    std::size_t result = std::numeric_limits<std::size_t>::max();
+
+    for (size_t i = 0; i < col.GetNumRows(); i++) {
+        if (col.IsNullOrEmpty(i)) continue;
+
+        auto const &string_data = mo::Type::GetValue<std::string>(col.GetValue(i));
+        std::istringstream iss(string_data);
+        std::vector<std::string> words_in_row(std::istream_iterator<std::string>{iss},
+                                              std::istream_iterator<std::string>());
+        std::size_t const &size = words_in_row.size();
+
+        if (size < result) result = size;
+    }
+
+    std::byte const *res = int_type.MakeValue(result);
+
+    return Statistic(res, &int_type, false);
+}
+
+Statistic DataStats::GetMaxWords(size_t index) const {
+    if (all_stats_[index].max_words.HasValue()) return all_stats_[index].max_words;
+    mo::TypedColumnData const &col = col_data_[index];
+    if (col.GetTypeId() != +mo::TypeId::kString) return {};
+
+    mo::IntType int_type;
+    std::size_t result = 0;
+
+    for (size_t i = 0; i < col.GetNumRows(); i++) {
+        if (col.IsNullOrEmpty(i)) continue;
+
+        auto const &string_data = mo::Type::GetValue<std::string>(col.GetValue(i));
+        std::istringstream iss(string_data);
+        std::vector<std::string> words_in_row(std::istream_iterator<std::string>{iss},
+                                              std::istream_iterator<std::string>());
+        std::size_t const &size = words_in_row.size();
+
+        if (size > result) result = size;
+    }
+
+    std::byte const *res = int_type.MakeValue(result);
+
+    return Statistic(res, &int_type, false);
+}
+
+Statistic DataStats::GetWordCount(size_t index) const {
+    if (all_stats_[index].word_count.HasValue()) return all_stats_[index].word_count;
+    mo::TypedColumnData const &col = col_data_[index];
+    if (col.GetTypeId() != +mo::TypeId::kString) return {};
+
+    mo::IntType int_type;
+    std::set<std::string> words;
+    std::size_t count = 0;
+    for (size_t i = 0; i < col.GetNumRows(); i++) {
+        if (col.IsNullOrEmpty(i)) continue;
+
+        auto const &string_data = mo::Type::GetValue<std::string>(col.GetValue(i));
+        std::istringstream iss(string_data);
+        std::vector<std::string> words_in_row(std::istream_iterator<std::string>{iss},
+                                              std::istream_iterator<std::string>());
```

Comment:
Create static function getWordsInString(str), use it here and in other places. Just count spaces in it.